viz_scripts/plots.py

```diff
@@ -19,115 +20,87 @@
+    # Concatenate DataFrames
+    df = pd.concat(all_data_frames, ignore_index=True)
 
-    m_labels, m_values = merge_small_entries(labels, values)
+    # Create DataFrame with unique combinations of 'Trip Type' and 'Mode'
+    unique_combinations = pd.DataFrame(list(product(df['Trip Type'].unique(), df['Mode'].unique())), columns=['Trip Type', 'Mode'])
 
-    wedges, texts, autotexts = ax.pie(m_values,
-                                      labels = m_labels,
-                                      colors=[colours[key] for key in labels],
-                                      pctdistance=0.75,
-                                      autopct= lambda pct: format_pct(pct, values),
-                                      textprops={'size': 23})
-
-    ax.set_title(plot_title, size=25)
-    plt.text(-1.3,-1.3,f"Last updated {arrow.get()}", fontsize=10)
-    plt.setp(autotexts, **{'color':'white', 'weight':'bold', 'fontsize':20})
-    plt.savefig(SAVE_DIR+file_name+".png", bbox_inches='tight')
-    plt.show()
+    # Merge the original DataFrame with the unique combinations DataFrame
+    merged_df = pd.merge(unique_combinations, df, on=['Trip Type', 'Mode'], how='left').fillna(0)
+    return merged_df
 
-def pie_chart_purpose(plot_title,labels,values,file_name):
-    
-    colours = dict(zip(labels, plt.cm.tab20.colors[:len(labels)]))
-    fig, ax = plt.subplots(figsize=(10, 10), subplot_kw=dict(aspect="equal"))
+def stacked_bar_chart_generic(plot_title, df, file_name, num_bars):
+    sns.set(font_scale=1.5)
+    fig, ax = plt.subplots(1, 1, figsize=(15, 6))
 
-    m_labels, m_values = merge_small_entries(labels, values)
-    
-    def func(pct, values):
-        total = sum(values)
-        absolute = int(round(pct*total/100.0))
-        return "{:.1f}%\n({:d})".format(pct, absolute) if pct > 3 else''
-    
-    wedges, texts, autotexts = ax.pie(m_values,
-                                      labels = m_labels,
-                                      colors=[colours[key] for key in labels],
-                                      pctdistance=0.85,
-                                      autopct=lambda pct: func(pct, values),
-                                      textprops={'size': 23})
-
-    ax.set_title(plot_title, size=25)
-    plt.text(-1.3,-1.3,f"Last updated {arrow.get()}", fontsize=10)
-    plt.setp(autotexts, **{'color':'white', 'weight':'bold', 'fontsize':20})
-    plt.savefig(SAVE_DIR+file_name+".png", bbox_inches='tight')
+    if num_bars == 1:
+        width = 2
+        ax.set_ylim(-0.4, 3)
+    else:
+        width = 0.8
+
+    running_total_long = [0] * num_bars
+
+    mode_mapping = {
+        "IN_VEHICLE": "IN_VEHICLE (Sensed)",
+        "UNKNOWN": "UNKNOWN (Sensed)",
+        "OTHER": "OTHER (Sensed)",
+        "BICYCLING": "BICYCLING (Sensed)",
+        "WALKING": "WALKING (Sensed)",
+        "AIR_OR_HSR": "AIR_OR_HSR (Sensed)"
+    }
+
+    colors = plt.cm.tab20.colors[:len(pd.unique(df['Mode']))]
```

Comment:
You probably already know this, but you need to ensure that the "unique colors" change is ported in here. This will not ensure that the colors are unique

Reply:
Incorporated use of subplots instead of combining different data frames into one.